core/drivers/serial8250_uart.c

```diff
@@ -86,7 +86,7 @@ static int serial8250_uart_getchar(struct serial_chip *chip)
 		/* Transmit FIFO is empty, waiting again */
 		;
 	}
-	return read8(base + UART_RHR);
+	return read32(base + UART_RHR);
```

Comment:
I wonder if we should mask the result with `0xff` just to make sure that this function doesn't return unexpected bits.